src/python/analyzer_executor/tests/test_cache_integration.py

```diff
@@ -0,0 +1,112 @@
+import unittest
+import pytest
+from faker import Faker
+
+from analyzer_executor_lib.analyzer_executor import AnalyzerExecutor
+
+SAMPLE_ADDR = "localhost"
+SAMPLE_PORT = "12345"
+
+
+@pytest.fixture
+def random_word():
+    fake = Faker()
+
+    def _random_word():
+        return "-".join(fake.words(nb=4))
+
+    return _random_word
+
+
```

Comment:
Personally:
i really, really hate monkeypatches

i would argue that i'd be happier if `AnalyzerExecutor.get_instance()` didn't inherently interact with `os.`, but instead was like

```
Environment = Mapping[str, Union[str, int]]

def get_instance(environment: Optional[Environment] = None) -> AnalyzerExecutor:
  environment = environment or os.environ
  # now, environment.get("key", default) to your heart's content
```
even if, technically, it exercises less of the true, production codepath

Reply:
For integration tests, I personally like to black box as much as possible.  To that end, I think my options are either monkeypatching, or rigging the literal OS env itself... and monkeypatching is simpler rn.

I agree with you about the long run though, I think we could stand to have env management in `grapl-common`.  This test straddles unit/integration boundaries and I'd like to have unit-level tests that say "given this config object, do we happy-path to Redis" and integration-level tests that say "give AE does its thing, expect keys XYZ to be inserted into Redis"

How about a ticket to centralize env management as a grapl-common lib, and then refactor the python services to use that?

Reply:
wrote self a reminder to ticket this (& other follow-on work like de-singleton) next week